Sources/NIO/BaseSocket.swift

```diff
@@ -58,9 +58,15 @@ import CNIOWindows
 #endif
 
 /// A Registration on a `Selector`, which is interested in an `SelectorEventSet`.
+/// `sequenceIdentifier` is used by the event notification backends (kqueue, epoll, ...)
+/// to mark events to allow for filtering of received return values to not be delivered to a
+/// new `Registration` instance that receives the same file descriptor. Ok if it wraps.
+/// Needed for i.e. testWeDoNotDeliverEventsForPreviouslyClosedChannels to succeed.
+typealias RegistrationSequenceIdentifier = UInt32
```

Comment:
Hmm, this typealias is causing me some confusion I think, because we're a bit inconsistent with how we use it. IMO we should either promote this to an actual type (`struct` wrapping a `UInt32`) or drop the typealias. We could aim to be consistent, but autocorrect has an awkward tendency to fail to tell you to use the typealias but instead to use the underlying type.